frontend/src/pages/SettingsPage/SettingsPage.tsx

```diff
@@ -115,6 +146,16 @@ export const SettingsPage = () => {
           <Divider />
           <Typography variant="subtitle1">Advanced</Typography>
           <List>
+            <SettingsListItem
+              label="Disable local network discovery"
+              subLabel={sublevelText}
+              icon={wifiConnectionLabel}
```

Comment:
We shouldn't need a local state for the icon, the preferences object will be updated right away.
```suggestion
              icon={preferences.disabledLocalNetwork ? 'wifi-slash' : 'wifi'}
```